stix2/datastore/taxii.py

```diff
@@ -83,6 +83,8 @@ def add(self, stix_data, version=None):
             # adding list of something - recurse on each
             for obj in stix_data:
                 self.add(obj, version=version)
+            else:
+                return  # If the list is empty just return.
```

Comment:
The else clause is executed if no `break` is encountered in the loop causing it to terminate early.  It isn't an empty check.  This loop body is a single line with no break, so it looks to me like the else clause could never be skipped.  It will always run.  So there may as well not be an else clause, right?  Wouldn't
```python
    for obj in stix_data:
        self.add(obj, version=version)

    return
```
do the same thing?

Reply:
I agree. I think the code is functioning correctly, but either the comment is outdated or it works for a different reason than what was intended.

Reply:
I think I can change it for what @chisholm recommends. The intention was to prevent the code to reach `self.collection.add_objects(bundle)` in the list case because bundle would be uninitialized.